pyiceberg/table/update/__init__.py

```diff
@@ -727,6 +727,14 @@ class AssertRefSnapshotId(ValidatableTableRequirement):
     ref: str = Field(...)
     snapshot_id: Optional[int] = Field(default=None, alias="snapshot-id")
 
+    @model_serializer
+    def ser_model(self) -> dict[str, Any]:
```

Comment:
nit: rename this to `serialize_model`, which aligns to the [pydantic example](https://docs.pydantic.dev/latest/api/functional_serializers/#pydantic.functional_serializers.model_serializer). makes this easier to find later 
```suggestion
    def serialize_model(self) -> dict[str, Any]:
```